src/components/Search/SearchRouter/SearchRouter.tsx

```diff
@@ -175,106 +183,119 @@ function SearchRouter({onRouterClose}: SearchRouterProps) {
+                    filteredAutocompleteSuggestions = filteredTypes.map((type) => ({filterKey: CONST.SEARCH.SYNTAX_ROOT_KEYS.TYPE, text: type}));
+                    break;
                 }
                 case CONST.SEARCH.SYNTAX_ROOT_KEYS.STATUS: {
                     const filteredStatuses = statusAutocompleteList
                         .filter((status) => status.includes(autocompleteValue.toLowerCase()) && !alreadyAutocompletedKeys.includes(status))
                         .sort()
                         .slice(0, 10);
-                    setAutocompleteSuggestions(filteredStatuses.map((status) => ({text: `${CONST.SEARCH.SYNTAX_ROOT_KEYS.STATUS}:${status}`, query: `${status}`})));
-                    return;
+                    filteredAutocompleteSuggestions = filteredStatuses.map((status) => ({filterKey: CONST.SEARCH.SYNTAX_ROOT_KEYS.STATUS, text: status}));
+                    break;
                 }
                 case CONST.SEARCH.SYNTAX_FILTER_KEYS.EXPENSE_TYPE: {
                     const filteredExpenseTypes = expenseTypes
                         .filter((expenseType) => expenseType.includes(autocompleteValue.toLowerCase()) && !alreadyAutocompletedKeys.includes(expenseType))
                         .sort();
-                    setAutocompleteSuggestions(
-                        filteredExpenseTypes.map((expenseType) => ({
-                            text: `${CONST.SEARCH.SYNTAX_FILTER_KEYS.EXPENSE_TYPE}:${expenseType}`,
-                            query: `${expenseType}`,
-                        })),
-                    );
-                    return;
+
+                    filteredAutocompleteSuggestions = filteredExpenseTypes.map((expenseType) => ({
+                        filterKey: CONST.SEARCH.SYNTAX_FILTER_KEYS.EXPENSE_TYPE,
+                        text: expenseType,
+                    }));
+                    break;
                 }
                 case CONST.SEARCH.SYNTAX_FILTER_KEYS.CARD_ID: {
                     const filteredCards = cardAutocompleteList
-                        .filter((card) => card.toLowerCase().includes(autocompleteValue.toLowerCase()) && !alreadyAutocompletedKeys.includes(card.toLowerCase()))
+                        .filter((card) => card.bank.toLowerCase().includes(autocompleteValue.toLowerCase()) && !alreadyAutocompletedKeys.includes(card.bank.toLowerCase()))
                         .sort()
                         .slice(0, 10);
-                    setAutocompleteSuggestions(
-                        filteredCards.map((card) => ({
-                            text: `${CONST.SEARCH.SYNTAX_FILTER_KEYS.CARD_ID}:${card}`,
-                            query: `${card}`,
-                        })),
-                    );
-                    return;
+
+                    filteredAutocompleteSuggestions = filteredCards.map((card) => ({
+                        filterKey: CONST.SEARCH.SYNTAX_FILTER_KEYS.CARD_ID,
+                        text: card.bank,
```

Comment:
Good point. I used the card generating function to not add new code and it looks nice

Reply:
Personally I like the idea of doing the last 4 digits with the dot separator: `Card name • 1234`

I _think_ we're leaning towards a similar solution for rooms that have the same name across workspaces (i.e. `#social`, `#admins`, etc.

Definitely want to hear from the rest of @Expensify/design though.

Also, does the query syntax for this need to be `cardID:` or could it just be `card:`?

Reply:
Interesting, for this idea @dannymcclain:

> Personally I like the idea of doing the last 4 digits with the dot separator:

The way I understand it is that if you don't give the card a nickname, the card's default name includes the last four. But if you give it a custom nickname, you can overwrite the card name and put whatever you want there.

I think I lean towards just doing something that uses the card name exactly as it appears in the cards table, so if there is a nickname, use that. But if not, use the default name (e.g. `Expensify Card - 1234`

This way we don't have to worry about any custom naming logic here, we just use whatever is in the table. Thoughts?

Reply:
Ahh yeah ok that makes sense! I didn't realize it worked that way. Totally down for that approach in that case!

Reply:
So I tried to copy the logic that generates names for on wallet Page, and here's how this looks (please ignore that there is more cards - this is fake dummy data):

![Screenshot 2024-11-04 at 16 37 17](https://github.com/user-attachments/assets/e871edb4-38e0-4535-8b22-1d23de1d9662)

Does this make more sense? As far as I understand correctly the logic on Wallet is as follows:
 - if we added a title to the card - use it
 - otherwise use property `card.bank` which I'm guessing would be the bank name.

There is no 4 last digits shown in this case, but I can add them.

Reply:
and @dannymcclain it is possible to changed `cardID:` into `card:` but it requires a bit of work and would best be coordinated with backend.
@luacmartins if making this change it would be best to switch it all across the grammar I think

Reply:
> There is no 4 last digits shown in this case, but I can add them.

What about the Expensify Cards table? Can you show us what some of those entries look like, especially ones without nicknames?

Reply:
@shawnborton sorry for my ignorance, but having never worked with the card related code I'm not sure I know how to properly simulate expensify cards 😅

Here are screenshots of 2 different methods of generating name
### 1. same as in wallet page
<img width="400" alt="screen-cards1" src="https://github.com/user-attachments/assets/c14cd861-f2d0-497f-b19d-bb626680626e">
### 2. using card `title` or `bank` 
<img width="400" alt="screen-cards2" src="https://github.com/user-attachments/assets/d62d826b-2ede-4a58-8704-e90a37b9abff">

The 3rd one from the top is what I believe is an "expensify card" since I set the bank name to `Expensify Bank`.
If that doesn't answer the question, then I need @luacmartins help

Reply:
Oh man I am torn here... the Wallet naming seems to make the most sense to me at first, but then we run into the issue where we use the generic name for Expensify Cards and thus all unnamed Expensify Cards would look the same right? Let's see if @luacmartins has any thoughts here.